utils/unittest/entity.go

```diff
@@ -0,0 +1,290 @@
+package unittest
+
+import (
+	"fmt"
+	"math/rand"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/onflow/crypto"
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// MockEntity implements a bare minimum entity for sake of test.
+type MockEntity struct {
+	Identifier flow.Identifier
+	Nonce      uint64
+}
+
+func (m MockEntity) ID() flow.Identifier {
+	return m.Identifier
+}
+
+func (m MockEntity) Checksum() flow.Identifier {
+	return m.Identifier
+}
+
+func EntityListFixture(n uint) []*MockEntity {
+	list := make([]*MockEntity, 0, n)
+
+	for i := uint(0); i < n; i++ {
+		list = append(list, &MockEntity{
+			Identifier: IdentifierFixture(),
+		})
+	}
+
+	return list
+}
+
+func MockEntityFixture() *MockEntity {
+	return &MockEntity{Identifier: IdentifierFixture()}
+}
+
+func MockEntityListFixture(count int) []*MockEntity {
+	entities := make([]*MockEntity, 0, count)
+	for i := 0; i < count; i++ {
+		entities = append(entities, MockEntityFixture())
+	}
+	return entities
+}
+
+// RequireEntityNonMalleable is a sanity check that the entity is not malleable with regards to the ID() function.
+// Non-malleability in this sense means that it is computationally hard to build a different entity with the same ID. This implies that in a non-malleable entity, changing any field should change the ID, which is checked by this function. Note that this is sanity check of non-malleability and that passing this test does not guarantee non-malleability.
```

Comment:
```suggestion
// Non-malleability in this sense means that it is computationally hard to build a different entity with the same ID.
// This implies that in a non-malleable entity, changing any field should change the ID, which is checked by this function.
// Note that this is sanity check of non-malleability and that passing this test does not guarantee non-malleability.
```
Breaking up a super-long line